numpy/core/src/common/half.hpp

```diff
@@ -72,7 +72,7 @@ class Half final {
     {
     #if defined(NPY_HAVE_AVX512FP16)
         __m128d md = _mm_load_sd(&f);
-        bits_ = static_cast<uint16_t>(_mm_cvtsi128_si32(_mm_castph_si128(_mm_cvtpd_ph(mf))));
+        bits_ = static_cast<uint16_t>(_mm_cvtsi128_si32(_mm_castph_si128(_mm_cvtpd_ph(md))));
```

Comment:
Ah, nvm. I guess this is a fix unrelated to the issue...

Reply:
`NPY_HAVE_AVX512FP16/NPY_HAVE_AVX512_SPR` seems to only supported by gcc-13

Reply:
IDK, why CI intel_spr_sde_test didn't trigger this error

Reply:
Ah, the missing SPR test issue bit us.

Reply:
`NPY_HAVE_AVX512_SPR ` is enabled with gcc >= 12.x. My nightly job picked up this failure with gcc-12 as well. See https://github.com/intel/x86-simd-sort/actions/runs/4879269275/jobs/8705679224.

The `intel_spr_sde_test` is disabled currently because of an SDE bug. I think what we could do is enable this CI just to build with gcc-12 without running the tests in the SDE.